cmd/aperture-agent/agent/otel-config.go

```diff
@@ -166,8 +166,13 @@ func addMetricsPipeline(
 	lis *listener.Listener,
 	promClient promapi.Client,
 ) {
+	pipeline := otelconfig.Pipeline{}
 	addPrometheusReceiver(config, agentConfig, tlsConfig, lis)
-	otelconfig.AddPrometheusRemoteWriteExporter(config, promClient)
+	if promClient != nil {
+		otelconfig.AddPrometheusRemoteWriteExporter(config, promClient)
+		pipeline.Exporters = []string{otelconsts.ExporterPrometheusRemoteWrite}
+	}
```

Comment:
Can you please check this with custom infra meters? We're using the same exporter `otelconsts.ExporterPrometheusRemoteWrite` there and I wonder if there's no problem with referring to non-existing exporter

Reply:
Made the changes to handle this